### PR TITLE
Added UI controls for Edge Cache Defensive Mode

### PR DESCRIPTION
**Pull request by fredrikekelund**
Fixes https://github.com/Automattic/dotcom-forge/issues/9077

## Proposed Changes

Introducing UI controls for Edge Cache Defensive Mode on the Server Settings page (`/hosting-config/:site_slug`)

| Edge cache disabled | Disabled | Enabled |
| - | - | - |
| edge-cache | disabled | enabled |

## Why are these changes being made?

See pcYYuD-1Sy-p2.

## Testing Instructions

1. Have an Atomic site
2. Navigate to `/hosting-config/:site_slug`
3. Disable edge cache
4. Ensure that the `Defensive mode` card contains the following text:

> Defensive mode requires [global edge cache](https://developer.wordpress.com/docs/site-performance/global-edge-cache/) to be enabled.

5. Enable edge cache
6. Enable edge cache defensive mode
7. Navigate to your website
8. Ensure that you see a brief `Checking your browser` page
9. Disable edge cache
10. Navigate to your website in a different browser
11. Ensure that you **do not** see a brief `Checking your browser` page

### CODE REVIEW

File: client/data/hosting/use-cache.ts [hunk 1]
```diff
@@ -88,42 +88,45 @@ export const useSetEdgeCacheMutation = (
 			queryClient.setQueryData( [ USE_EDGE_CACHE_QUERY_KEY, siteId ], active );
 			return previousData;
 		},
-		onSuccess( ...args ) {
-			const [ , { active } ] = args;
```
Comment (fredrikekelund) [inline client/data/hosting/use-cache.ts:92]: This change obviously isn't necessary, but I chose to do it for a few reasons:
1. It's more readable.
2. This is consistent with my new `onSuccess` implementation in `useEdgeCacheDefensiveModeMutation`.
3. We have type checking for `options.onSuccess`, so Typescript will complain if we pass the wrong or too few arguments.

File: client/data/hosting/use-cache.ts [hunk 2]
```diff
@@ -48,14 +51,11 @@ export const purgeEdgeCache = async ( siteId: number ) => {
 	} );
 };
 
-export const useEdgeCacheQuery = ( siteId: number ) => {
-	return useQuery< boolean, unknown, boolean >( {
+export const useEdgeCacheQuery = ( siteId: number | null ) => {
+	return useQuery< boolean >( {
 		queryKey: [ USE_EDGE_CACHE_QUERY_KEY, siteId ],
-		queryFn: () => getEdgeCacheStatus( siteId ),
+		queryFn: () => getEdgeCacheStatus( siteId as number ),
```
Comment (fredrikekelund) [inline client/data/hosting/use-cache.ts:57]: Not super happy with this `as` declaration, but the `enabled: !! siteId` directive should make it safe, at least.

File: client/hosting/server-settings/components/defensive-mode-card/index.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+	const enabledUntil = moment.unix( defensiveModeData?.enabled_until ?? 0 );
+
+	return (
+		<HostingCard className="defensive-mode-card" title={ translate( 'Defensive mode' ) }>
+			<HostingCardDescription>
+				{ translate(
+					'Extra protection against spam bots and attacks. Visitors will see a quick loading page while we run additional security checks. {{a}}Learn more{{/a}}',
+					{
+						components: {
+							a: <InlineSupportLink supportContext="hosting-defensive-mode" showIcon={ false } />,
```
Comment (matt-west) [inline client/hosting/server-settings/components/defensive-mode-card/index.tsx:73]: The `Learn more` link opened the Global Edge Cache support doc, but I would expect to see something that explains what defensive mode is and how to enable/disable it.

File: client/hosting/server-settings/components/defensive-mode-card/index.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+import './style.scss';
+
+type DefensiveModeCardProps = {
+	disabled?: boolean;
+};
+
+const AVAILABLE_TTLS = [
+	{
+		label: '1h',
```
Comment (matt-west) [inline client/hosting/server-settings/components/defensive-mode-card/index.tsx]: Can we write these labels in full. (i.e. `1 hours`, `2 days`, etc.)

File: client/hosting/server-settings/components/defensive-mode-card/style.scss [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+	align-items: center;
+	display: flex;
+	gap: 10px;
+
+	p {
+		margin: 0;
+	}
+}
+
+.defensive-mode-card__enabled-indicator {
```
Comment (matt-west) [inline client/hosting/server-settings/components/defensive-mode-card/style.scss:25]: This indicator should pulse. See the [Figma component](https://www.figma.com/design/kvZMElckk83g77JSRWOdU8/Server-Settings-Enhancements-i1?node-id=1907-8155&m=dev) for states.

If we set a transparent border to begin with, we should be able to animate the border color to achieve the desired effect.

**Review by matt-west — CHANGES_REQUESTED** [review]
Thanks @fredrikekelund. This is looking pretty good.

I added a few comments.

File: client/hosting/server-settings/components/defensive-mode-card/index.tsx (hunk 3 above)
Comment (fredrikekelund) [reply client/hosting/server-settings/components/defensive-mode-card/index.tsx:73]: Yup, good point. We have https://github.com/Automattic/dotcom-forge/issues/9126 to address this.

@alexapeduzzi, do you think some docs about defensive mode fit into pfA0Z5-70-p2, or should we spin up a separate request for that?

File: client/hosting/server-settings/components/defensive-mode-card/style.scss (hunk 5 above)
Comment (fredrikekelund) [reply client/hosting/server-settings/components/defensive-mode-card/style.scss:25]: Ah, thanks. I'll look into it!

Comment (matt-west) [reply client/hosting/server-settings/components/defensive-mode-card/style.scss:25]: Let me know if you need any help with the animation.

Comment (fredrikekelund) [reply client/hosting/server-settings/components/defensive-mode-card/style.scss:25]: @matt-west, I've pushed a change here. Let me know your thoughts! Feel free to suggest a tweak to the CSS if you want

Comment (matt-west) [reply client/hosting/server-settings/components/defensive-mode-card/style.scss:25]: Let’s slow the animation to `1.5s`, but other than that it looks great. Thanks @fredrikekelund!

**Review by matt-west — APPROVED** [review]

**Comment by wojtekn** [comment]
@fredrikekelund feature-wise it works well for me. I needed to disable the proxy to test it. The only minor thing I noticed its the date format. We use this one in defensive mode block:

> 2:13 PM on September 26, 2024.

In the last backup time block we use this:
> September 26, 2024 6:25 AM

And time in logs tab:
> Sep 23, 2024 at 12:29 PM

We are not fully consistent in other places, but would it make sense to use one of the formats we use in other tabs already?

**Review by wojtekn — APPROVED** [review]

**Comment by fredrikekelund** [comment]
@wojtekn, good point about the date format 👍 I went with the `LLL` format used in the last backup card.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16882963

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @fredrikekelund for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16882963).

**Pull request merged**